homeassistant/components/hyundai_kia_connect/binary_sensor.py

```diff
@@ -0,0 +1,206 @@
+"""Sensor for Hyundai / Kia Connect integration."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+import logging
+from typing import Final
+
+from hyundai_kia_connect_api import Vehicle
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+
+from .const import DOMAIN
+from .entity import HyundaiKiaConnectEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class HyundaiKiaBinarySensorEntityDescription(BinarySensorEntityDescription):
+    """A class that describes custom binary sensor entities."""
+
+    on_icon: str | None = None
+    off_icon: str | None = None
+
+
+SENSOR_DESCRIPTIONS: Final[tuple[HyundaiKiaBinarySensorEntityDescription, ...]] = (
+    HyundaiKiaBinarySensorEntityDescription(
+        key="engine_is_running",
+        name="Engine",
+        device_class=BinarySensorDeviceClass.POWER,
+        on_icon="mdi:engine",
+        off_icon="mdi:engine-off",
```

Comment:
When using a device class, we should not set an icon.

```suggestion
```

Reply:
@frenck , having a device class enables specific icons, but it is not limited to it. Name of states and trigger events are evolving around device classes too. Rather than `X becomes on`, it is visible to user as `X is Open`. But using door icons do not add any value to this use case. So, if there is no explicit reason not to mix device classes with icons or architectural decisions, I prefer to keep it as it enhances user experience. Otherwise, I prefer to drop device classes and stick these icons.